qiskit_experiments/analysis/guesses.py

```diff
@@ -0,0 +1,189 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+A library of parameter guess functions.
+"""
+# pylint: disable=invalid-name
+
+from typing import Optional, Tuple
+
+import numpy as np
+from scipy import signal
+
+
+def frequency(x: np.ndarray, y: np.ndarray) -> float:
+    """Get frequency of oscillating signal.
+
+    .. note::
+
+        This function returns always positive frequency.
+
+    Args:
+        x: Array of x values.
+        y: Array of y values.
+
+    Returns:
+        Frequency estimation of oscillation signal.
+    """
+    fft_data = np.fft.fft(y - np.average(y))
+    sampling_rate = float(np.mean(np.diff(x)))
+    freqs = np.fft.fftfreq(len(x), sampling_rate)
+
+    positive_freqs = freqs[freqs >= 0]
+    positive_fft_data = fft_data[freqs >= 0]
+
+    return positive_freqs[np.argmax(np.abs(positive_fft_data))]
+
+
+def max_height(
+    y: np.ndarray,
+    percentile: Optional[float] = None,
+    absolute: bool = False,
+) -> Tuple[float, int]:
+    """Get maximum value of y curve and its index.
+
+    Args:
+        y: Array of y values.
+        percentile: Return that percentile value if provided, otherwise just return max value.
+        absolute: Use absolute y value.
+
+    Returns:
+        The maximum y value and index.
+    """
+    if absolute:
+        y_ = np.abs(y)
+    else:
+        y_ = y
+
+    if percentile is not None:
+        y_max = np.percentile(y_, percentile)
+    else:
+        y_max = np.max(y_)
+
+    index = list(y_).index(y_max)
+
+    return y_max, index
+
+
+def min_height(
+    y: np.ndarray,
+    percentile: Optional[float] = None,
+    absolute: bool = False,
+) -> Tuple[float, int]:
+    """Get minimum value of y curve and its index.
+
+    Args:
+        y: Array of y values.
+        percentile: Return that percentile value if provided, otherwise just return min value.
+        absolute: Use absolute y value.
+
+    Returns:
+        The minimum y value and index.
+    """
+    if absolute:
+        y_ = np.abs(y)
+    else:
+        y_ = y
+
+    if percentile is not None:
+        y_min = np.percentile(y_, percentile)
+    else:
+        y_min = np.min(y_)
+
+    index = list(y_).index(y_min)
+
+    return y_min, index
+
```

Comment:
Consider writing an internal function that will be called by `max_height` and `min_height`

Reply:
A general `get_height` method is added to remove duplicate code https://github.com/Qiskit/qiskit-experiments/pull/154/commits/63ed1c319737890fd4459f34de0266ffaecd970e